compiler/runtime.go

```diff
@@ -7,8 +7,25 @@ import (
 	"github.com/influxdata/flux/ast"
 	"github.com/influxdata/flux/semantic"
 	"github.com/influxdata/flux/values"
+	"github.com/pkg/errors"
 )
 
+type Func interface {
```

Comment:
This change does not modify the Func interface except for where types were concerned. 

I am not sure I follow the concern about type casting, the point of this interface is so that you do not have to type cast at all. The `compiler` follows the types through and handles concrete types during the flow of data where it can. The `scope` being the obvious exception.

Reply:
Ah, I did not see this was moved code.

My comment still applies, but in a more general way and isn't directly related to this PR. I thought this was a new interface.